runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -34,6 +34,44 @@
 #include "backend/controlflow/KernelGenerator.h"
 #include <memory>
 
+namespace onert
+{
+namespace
+{
+
+class SyncFunction final : public exec::IFunction
+{
+public:
+  virtual ~SyncFunction() = default;
+  SyncFunction(std::unique_ptr<exec::IFunction> fn, const std::shared_ptr<backend::IConfig> config)
+      : _fn{std::move(fn)}, _config{config}
+  {
+    assert(_fn);
+    assert(_config);
+  }
+
+  void run() override
+  {
+    _fn->run();
+    _config->sync();
```

Comment:
Always sync?

Reply:
Yes. That's why its name is `SyncFunction`. `runSync` will be removed.